Fix acr npe when binarydata is not buffered

lmolkova opened the pull request.

Commit:
Remove ACR redirct policy and add content-length sanity check (#34737)

* Remove ACR redirct policy and add content-length sanity check

kushagraThapar committed:
Reduced timeouts on tests with unecessary high values (#34749)

Commit:
fix acr npe when binarydata is not buffered

kushagraThapar committed:
Reduced timeouts on tests with unecessary high values (#34758)

srnagar commented on sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/UtilsImpl.java:
```diff
@@ -466,6 +466,14 @@ public static long getBlobSize(HttpHeader contentRangeHeader) {
+            return Long.parseLong(contentLengthHeader.getValue());
+        }
+
+        throw LOGGER.logExceptionAsError(new ServiceResponseException("Content-Length header in missing in the response"));
```
Are there responses where we've noticed service not returning content length? I am expecting that this would never happen but good to handle this case. Should we throw an exception (resulting in a bit of a behavior change) or should we fall back to the existing behavior of  getting the length from BinaryData (i.e. fallback to buffering)?

lmolkova replied:
Good point! 
We're not using transfer-encoding, so service [SHOULD send](https://www.rfc-editor.org/rfc/rfc9110.html#name-content-length) content-length. So it'd indicate a bug on the service side and we've been told that that container registry should not be trusted and we should fail fast if unexpected things happen. 
This behavior was never released in a stable version, so it's ok to break.

srnagar replied:
Makes sense to throw an exception then.

mkrueger committed:
Corrected field naming in documentation. (#34759)

Fixes #34736

kushagraThapar committed:
Updated code to use Java UUID instead of fasterxml (#34747)

mbhaskar committed:
End end timeout policy (#34554)

* end-end timeout implementation
* Refactoring cache refresh logic

asaflevi-ms committed:
[healthinsights] support live tests. (#34727)

[healthinsights] support live tests.

alzimmermsft committed:
Add JsonSerializable Support to Azure Core JacksonAdapter (#34742)

Add JsonSerializable Handling into Core JacksonAdapter

juntuchen-msft committed:
callMedia methods update (#34752)

* callMedia methods update

* fixed pipeline

Commit:
Add jackson 2.15 to dependency test matrix (#34755)

* add jackson 2.15 to test matrix

huachuandeng committed:
remove callinvite from transfer (#34754)

* remove callinvite from transfer

* fix spotbug

samvaity committed:
[Test Proxy] Enable setting recording options and reset proxy data (#34704)

abbytec committed:
Update IncrementalChangeFeedProcessorTest.java (#34726)

fixed noncompliant test code
https://rules.sonarsource.com/java/tag/assertj/RSPEC-5833

alzimmermsft committed:
Ignore Service Version Query Parameter in Playback (#34761)

minnieliu committed:
Clean up readme and change log (#34753)

* Clean up readme and change log

* Fix compilation

* Address comments

* Address comments

* Fix build ci pipeline

---------

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

azure-sdk committed:
pass along the artifact name so we can override it later (#34767)

Co-authored-by: scbedd <45376673+scbedd@users.noreply.github.com>

azure-sdk committed:
Update azure-sdk-build-tools Repository Resource Refs in Yaml files (#34658)

samvaity committed:
Fix bug when setting table cell bounding box values (#34769)

Commit:
Add session idle timeout (#34700)

* Add session idle timeout config option

anuchandy committed:
[Messaging] Reduce allocation for Send by using RedableBuffer (#34591)

* Using RedableBuffer to send

* better fun name, revert collapsed imports

* Using rewind to seek the position when retrying

Co-authored-by: Liudmila Molkova <limolkova@microsoft.com>

* Productize the ReadableBuffer based apporach

* remove var added for dev-time testing

* Moving code comment to git description

* changelog update

* let's not new-up ReadableBuffer in non-batch cases

* (review feedback): use monoError to report buffer-overflow and log when a message in the batch is empty.

---------

Co-authored-by: Liudmila Molkova <limolkova@microsoft.com>

azure-sdk committed:
Increment package versions for paloaltonetworks releases (#34772)

alzimmermsft committed:
Include Long-Running Tests in Thread Dump (#34374)

srnagar committed:
Update RestProxy to not buffer for streaming response (#34750)

* Update RestProxy to not buffer for streaming response

* unit tests

* remove unused imports
